packs/elastic/actions/search.py

```diff
@@ -0,0 +1,87 @@
+from easydict import EasyDict
+from lib.items_selector import ItemsSelector
+from lib.elastic_action import ElasticAction
+from collections import defaultdict
+import logging
+import sys
+import elasticsearch
+import json
+
+logger = logging.getLogger(__name__)
+
+
+class SearchRunner(ElasticAction):
+
+    def __init__(self, config=None):
+        super(SearchRunner, self).__init__(config=config)
+        self._iselector = None
+
+
+    @property
+    def iselector(self):
+        """
+        Used to fetch indices/snapshots and apply filter to them.
+        """
+        if not self._iselector:
+            self._iselector = ItemsSelector(self.client, **self.config)
+        return self._iselector
+
+
+    def run(self, **kwargs):
```

Comment:
I see.

In this case it might make more sense to move common functionality into a base class and have two classes (one for each action) inherit from it - this way each class can implement `run` method with the respective arguments.

The whole idea of Python runner actions is that they are pretty much Python scripts / classes which can be re-used (or tested) outside of StackStorm action runner. Doing that `**kwargs` magic breaks this assumption since those scripts are now highly coupled to the metadata file and those scripts can't be easily re-used or tested without looking at the metadata file.